numba/targets/arraymath.py

```diff
@@ -268,8 +226,129 @@ def array_sum_impl_axis(arr, axis):
                 elif axis == 3:
                     index_tuple4 = _gen_index_tuple(arr.shape, axis_index, 3)
                     result += arr[index_tuple4]
-
         return op(result, 0)
+    return inner
+
+
+@lower_builtin(np.sum, types.Array, types.intp, types.DTypeSpec)
+@lower_builtin(np.sum, types.Array, types.IntegerLiteral, types.DTypeSpec)
+@lower_builtin("array.sum", types.Array, types.intp, types.DTypeSpec)
+@lower_builtin("array.sum", types.Array, types.IntegerLiteral, types.DTypeSpec)
+def array_sum_axis_dtype(context, builder, sig, args):
+    """
+    The third parameter to gen_index_tuple that generates the indexing
```

Comment:
This comment doesn't appear to be related to the code around it. I am not sure even if it is still needed. It didn't help me understand the code better.

Reply:
thanks, there was refactoring and this comment should have moved with `gen_sum_axis_impl`. I did that now.

Reply:
That is what a good review is for. 😄